lta/transfer/globus.py

```diff
@@ -1,108 +1,196 @@
+        """Wait (forever) for transfer to finish.
+
+        Return:
+            The successful task object.
+
+        Raises:
+            'GlobusTransferFailedException' if the transfer failed (FAILED or INACTIVE).
+
+        NOTE: 'globus_sdk.TransferClient.task_wait()' is *NOT* async, so we must diy
+        """
+        first = True
+        while True:
+
+            # sleep each time after first iteration
+            if first:
+                first = False
             else:
-                cmd = ['grid-proxy-init']
-            cmd.extend(['-debug', '-pwstdin', '-valid', f'{int(self.cfg["GLOBUS_PROXY_DURATION"])+1}:0'])
-            if 'GLOBUS_PROXY_OUTPUT' in self.cfg and self.cfg['GLOBUS_PROXY_OUTPUT']:
-                cmd.extend(['-out', cast(str, self.cfg['GLOBUS_PROXY_OUTPUT'])])
-            inputbytes = (cast(str, self.cfg['GLOBUS_PROXY_PASSPHRASE']) + '\n').encode('utf-8')
-            p = subprocess.run(cmd, input=inputbytes, capture_output=True, timeout=60, check=False)
-            logger.info('proxy cmd: %r', p.args)
-            logger.info('stdout: %s', p.stdout)
-            logger.info('stderr: %s', p.stderr)
-            if 'GLOBUS_PROXY_VOMS_VO' in self.cfg and self.cfg['GLOBUS_PROXY_VOMS_VO']:
-                for line in p.stdout.decode('utf-8').split('\n'):
-                    if line.startswith('Creating proxy') and line.endswith('Done'):
-                        break  # this is a good proxy
-                else:
-                    raise Exception('voms-proxy-init failed')
-            elif p.returncode > 0:
-                raise Exception('grid-proxy-init failed')
-
-    def get_proxy(self) -> Any:
-        """Get the proxy location."""
-        if 'GLOBUS_PROXY_OUTPUT' in self.cfg and self.cfg['GLOBUS_PROXY_OUTPUT']:
-            return self.cfg['GLOBUS_PROXY_OUTPUT']
-        FNULL = open(os.devnull, 'w')
-        return subprocess.check_output(['grid-proxy-info', '-path'],
-                                       stderr=FNULL).decode('utf-8').strip()
+                await asyncio.sleep(self._env.GLOBUS_POLL_INTERVAL_SECONDS)
+
+            # look at status
+            LOGGER.debug("checking transfer status...")
+            task = self._transfer_client.get_task(task_id)
+            LOGGER.debug(f"{task=}")
+            status = task["status"]
+            LOGGER.debug(f"{status=}")
+            match status:
+                case "SUCCEEDED":
+                    # File(s) in transfer was/were:
+                    #   (1) written at destination and/or
+                    #   (2) did not need to be written due to `sync_level` setting
+                    # To differentiate, look at `task` fields:
+                    #   'files' (int),
+                    #   'files_skipped' (int), and
+                    #   'files_transferred' (int)
+                    LOGGER.info(f"Globus transfer succeeded: {task_id=}")
+                    return task.data
+                case "FAILED" | "INACTIVE":
```

Comment:
I was worried that `INACTIVE` might be delayed transfers (i.e. too many in parallel to be active at once); but the docs have corrected me:

>"INACTIVE"
>The task has been suspended and will not continue without intervention. Currently, only credential expiration will cause this state.

I think you're handling the transfer statuses correctly here.